File: backend/src/main/java/io/github/nicolasdesnoust/lambda/contact/domain/ContactMessage.java
package io.github.nicolasdesnoust.lambda.contact.domain;

import lombok.Builder;
import lombok.Value;

@Value
@Builder(setterPrefix = "with")
public class ContactMessage {

    String subject;
    String body;
    Sender sender;

    @Value
    @Builder(setterPrefix = "with")
    public static class Sender {

        String name;
        String emailAddress;

    }
}

File: backend/functions/send-contact-mail/src/domain/contact-message.ts
import { Sender } from "./sender";

export interface ContactMessageProps {
  subject: string;
  body: string;
  sender: Sender;
}

export class ContactMessage {
  private _props: ContactMessageProps;

  constructor(props: ContactMessageProps) {
    this._props = props;
  }

  public get subject(): string {
    return this._props.subject;
  }

  public get body(): string {
    return this._props.body;
  }

  public get sender(): Sender {
    return this._props.sender;
  }
}
